docs/react-v9/contributing/rfcs/react-components/convergence/z-index-handling.md

```diff
@@ -0,0 +1,314 @@
+
+### What will be covered
+
+This proposal utilize this elevation system to define layers that translate into z-axis values, and those can be used by the components. The concepts are:
+
+- Each layer have a well-thought z-index number mapped to it
+- Multiple components can occupy the same layer, if they share the same importance level in the stack (e.g. Drawer and Nav)
+- A component can be defined without a z-index. Its placement fallbacks to its position within the [stack context](https://web.dev/learn/css/z-index/#stacking-context).
+
+### What won't be covered
+
+- Negative z-indexes. Components can define negative z-index values to represent background objects. Since this is very subjective, it'll be up to the component to define its own negative z-index values.
+- Correlation between shadows and z-indexes. Even though it might be related, this is out of the scope of this proposal. If necessary, it can be covered by a follow-up RFC.
+
+## Implementation
+
+> [!NOTE]
+> The approach here only standardize the layers and define z-index values for them. This is great for the current state of our components that can define arbitrary values, but won't solve a very specific problem: Defining priority for similar UI elements. e.g. Two Dropdowns are created. Which of them should have higher priority and therefore be displayed on top?
+
+Expose the z-index values as tokens. This would map the layers into token values that can be used anywhere in styles.
+
+The layers can be defined as follows:
+
+```ts
+// packages/tokens/src/global/zIndexes.ts
+
+import { ZIndexLevels, ZIndexTokens } from '../types';
+
+// According to Fluent V2 guidelines
+const levels: ZIndexLevels = [
+  0, // Elevation 0
+  1, // Elevation 2
+  1000, // Elevation 4
+  2000, // Elevation 8
+  3000, // Elevation 16
+  4000, // Elevation 28
+  5000, // Elevation 64
+  6000, // High priority elements
+];
+
+/**
+ * Global z-index values for elements
+ */
+export const zIndexes: ZIndexTokens = {
+  zIndexBackground: levels[0], // default
+  zIndexContent: levels[1], // content - header, footer, sidebar
+  zIndexOverlay: levels[2], // overlay elements - drawer, nav
+  zIndexPopup: levels[3], // popup layers - popups, modals, dialogs
+  zIndexMessages: levels[4], // communication elements - banners, messages, toasts, snackbar
+  zIndexFloating: levels[5], // floating elements - dropdowns, teaching
+  zIndexPriority: levels[6], // priority elements - tooltips
+  zIndexDebug: levels[7], // debug - error overlays, debug messages
+};
+```
+
+```ts
+// Fluent/partner code
+
+import { tokens } from '@fluentui/theme';
+
+const styles = {
+  root: {
+    zIndex: tokens.zIndexOverlay /* var(--zIndexOverlay) */,
+  },
+};
+```
```

Comment:
Can you please also:
- include changes to types i.e. `Theme` in `packages/tokens/src/types.ts` 
- add an example of customization i.e.
  ```tsx
  const customTheme: Theme = {
    ...webLightTheme,
    // 👇 customized values
    zIndexOverlay: 100000
  }

  function App() {
    return <FluentProvider theme={customTheme} />
  }
  ```